org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -5971,6 +5972,68 @@ public static void run(IWorkspaceRunnable action, ISchedulingRule rule, IProgres
 			workspace.run(new BatchOperation(action), rule, IWorkspace.AVOID_UPDATE, monitor);
 		}
 	}
+	/**
+	 * @since 3.37
+	 */
+	@FunctionalInterface
+	public static interface JavaCallable<V, E extends Exception> {
+		/**
+		 * Computes a value or throws an exception.
+		 *
+		 * @return the result
+		 * @throws E the Exception of given type
+		 */
+		V call() throws E;
+	}
+	/**
+	 * @since 3.37
+	 */
+	@FunctionalInterface
+	public static interface JavaRunnable<E extends Exception> {
```

Comment:
`JavaCoreRunnable`? `JDTRunnable`? `JavaModelRunnable`?

`JavaCore` is not much better than `JavaRunnable` as a name, so I'm fine with the name here. But maybe some better name can be invented for the new code here? Same for the one above.

Reply:
Since these are @FunctionalInterface their names will unlikely be ever written. Typically it will be just a lambda. so name does not matter here